diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d0c0f1..9b74a85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,9 @@ find_library(CEDAR_MEMADAPTER_LIBRARY NAMES MemAdapter)
 find_library(CEDAR_VE_LIBRARY NAMES VE)
 find_library(CEDAR_VIDEOENGINE_LIBRARY NAMES videoengine)
 find_library(ROCKCHIP_MPP_LIBRARY NAMES rockchip_mpp)
+if(NOT ROCKCHIP_MPP_LIBRARY)
+    message(FATAL_ERROR "ROCKNIX openhd-glide requires Rockchip MPP")
+endif()
 
 set(LV_CONF_PATH ${CMAKE_CURRENT_SOURCE_DIR}/src/glide_ui/lv_conf.h CACHE STRING "" FORCE)
 set(LV_CONF_BUILD_DISABLE_EXAMPLES ON CACHE BOOL "" FORCE)
