diff --git a/CMakeLists.txt b/CMakeLists.txt
index 263d19c8..d809c727 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMakeTargets")
 # ----------------------------------------------------------------------------
 # enable test in this project
 # ----------------------------------------------------------------------------
-option(BUILD_TEST "enable test binary building)" ON)
+option(BUILD_TEST "enable test binary building)" OFF)
 # ----------------------------------------------------------------------------
 # export json compile commands
 # ----------------------------------------------------------------------------
@@ -340,7 +340,7 @@ include_directories(kmpp/base/inc)
 # ----------------------------------------------------------------------------
 #  utils for test case
 # ----------------------------------------------------------------------------
-add_subdirectory(utils)
+#add_subdirectory(utils)
 
 # ----------------------------------------------------------------------------
 #  Kernel Media Process Platform library
@@ -356,7 +356,7 @@ add_subdirectory(mpp)
 # ----------------------------------------------------------------------------
 #  test / demo
 # ----------------------------------------------------------------------------
-add_subdirectory(test)
+#add_subdirectory(test)
 
 # ----------------------------------------------------------------------------
 #  install headers
